docs/js/class_p.js

```diff
@@ -9337,15 +9137,25 @@ class Puzzle {
                     this.redraw();
                 }
             } else {
-                if (!this[this.mode.qa].surface[num] || this[this.mode.qa].surface[num] != this.drawing_mode) {
+                let cc = undefined;
+                if (UserSettings.custom_colors_on) {
+                    // If left click second time (i.e. DG option) and moving or right click and moving
+                    //ML: Why an exception for green (2) ??? 
+                    //if (this.drawing_mode === 2 || this.mouse_click === 2) { 
+                    if (this.mouse_click === 2) {
+                        cc = this.get_rgbcolor(this.drawing_mode);
+                    } else {
+                        cc = this.get_customcolor();
+                    }
+                }
+                if (!this[this.mode.qa].surface[num] || this[this.mode.qa].surface[num] != this.drawing_mode || this[this.mode.qa + "_col"].surface[num] != cc) {

```

Comment:
I traced down the bug to this. The new third condition "this[this.mode.qa + "_col"].surface[num] != cc" continuous to remain true will hovering over the same cell after mouse down and so it keeps recording the event.
Removing this fixes the issue.
Do you remember why you needed this third condition?

Reply:
Sounds good. Thanks for the explanation. Yeah, I noticed the the comparing of "undefined" with cc and triggering it true.